train_MF.py

```
@@ -0,0 +1,183 @@
+from preprocessing.create_batch_dataset import PDB_Dataset
+from tqdm import tqdm  # FIX the import
+from utils import calculate_class_weights, save_alpha_weights, load_alpha_weights
+from torch.optim.lr_scheduler import CosineAnnealingLR
+import os
+
+
+# Load datasets from the pickle file
+dataset_save_path = "/home/hpc_users/2019s17273@stu.cmb.ac.lk/ganeshiny/protein-go-predictor/preprocessing/data/sachinthadata/MFO_datasets.pkl"
+
+with open(dataset_save_path, 'rb') as f:
+    datasets = pickle.load(f)
```

Ruff (0.12.2)

`pickle` and modules that wrap it can be unsafe when used to deserialize untrusted data, possible security issue

(S301)